CMakeLists.txt

```diff
@@ -11,6 +11,9 @@ project(
 
 enable_testing()
 
+include(cmake/feature_test.cmake)
```

Comment:
@neatudarius Do you expect we'll be moving utilities like this into https://github.com/bemanproject/infra at some point? Or should we spin out another project for reusable CMake bits?

Reply:
I am designing this so it would be easily packagable into a CMake module. I would prefer putting this into infra for now.

Reply:
Sounds good. I agree.